docs/modules/ROOT/pages/headers.adoc

```diff
@@ -0,0 +1,54 @@
+# DataSonnet Header
+
+The header is an optional comment section in the mapping document which must start with `/\** DataSonnet` followed by a new line character and end with a line `*/` and contains a set of configuration properties.
+The properties are hierarchical, with sections separated by the `.` (dot) character, e.g.:
+
+------------
+dataformat.application/xml.payload.NamespaceSeparator=:
+dataformat.application/xml.payload.TextValueKey=__text
+dataformat.application/csv.myVar.Quote="
+dataformat.application/csv.myVar.Separator=|
+------------
+
+The header must be the first comment before any other line of code, in the following format:
+
+-------------
+/** DataSonnet
+version=<VERSION_NUMBER>
+<PROPERTY>=<VALUE>
+...
+*/
+-------------
+*Example*
+------------------------
+/** DataSonnet
+version=1.0
+*/
+//DataSonnet mapping starts here
+{
+    mappedValue: payload.value
+}
+------------------------
+
+The header is optional; if omitted, all default settings will be used.
+
+## Data Format Properties
+
+The data format properties control the data format plugins behavior. The property names must start with `dataformat.<MIME_TYPE>.payload|output|*|<VAR_NAME>` where `<MIME_TYPE>` is the mime type supported by the DataSonnet. If a property is not recognized, an error will be raised.
```

Comment:
I'm not on board with "output" as a special value indicating the output. Payload works because it's a name that can't be used for any other input name due to being reserved for the main input, but someone could, for whatever reason, name an input "output".

Also, it makes it harder to read the property section if the output isn't in some way separated.

Let me see, some options... 

`input.name.mime/type.param = `

`output.mime/type.param =`

?

On a side note, one thing we'll want to keep an eye on is that output params may turn out to be something that's of interest to override at runtime. Doesn't really affect the calculus here, though.

Reply:
Hmm, yeah, agree, I'll make the changes.

Reply:
Actually - it's not that simple :) Let's have a call and discuss.